packages/fx-core/src/common/local/localEnvManager.ts

```diff
@@ -105,4 +133,28 @@ export class LocalEnvManager {
     }
     throw error;
   }
+
+  private getValidDeps(projectSettings: ProjectSettings): DepsType[] {
+    const depsTypes: DepsType[] = [];
+
+    depsTypes.push(DepsType.AzureNode);
+    if (ProjectSettingsHelper.includeFrontend(projectSettings)) {
+      depsTypes.push(DepsType.Dotnet);
+    }
+
+    if (ProjectSettingsHelper.includeBackend(projectSettings)) {
+      depsTypes.push(DepsType.FunctionNode);
+      depsTypes.push(DepsType.FuncCoreTools);
+    }
+
+    if (ProjectSettingsHelper.includeBot(projectSettings)) {
+      depsTypes.push(DepsType.Ngrok);
+    }
+
+    if (ProjectSettingsHelper.isSpfx(projectSettings)) {
```

Comment:
From my understanding, for Azure project, only AzureNode checker is needed and for SPFx projects, only SpfxNode checker is needed.

Reply:
Remove node checker and will implement it in another PR.